python-package/lets_plot/plot/plot.py

```diff
@@ -14,61 +14,72 @@
-                useful when multiple data frames are used to produce different layers, as is often the case in complex
-                graphics.
-
-        ggplot() with no layers defined will produce an error message: "No layers in plot"
+    ----
+    ggplot() initializes a ggplot object. It can be used to declare the input data frame for a graphic and to
+    specify the set of plot aesthetics intended to be common throughout all subsequent layers unless specifically
+    overridden.
+
+    ggplot() is typically used to construct a plot incrementally, using the + operator to add layers to the
+    existing ggplot object. This is advantageous in that the code is explicit about which layers are added
+    and the order in which they are added. For complex graphics with multiple layers, initialization with
+    ggplot() is recommended.
+
+    There are three common ways to invoke ggplot (see examples below):
+
+    - ggplot(data, aes(x,y)):
+      This method is recommended if all layers use the same data and the same set of aesthetics, although
+      this method can also be used to add a layer using data from another data frame.
+
+    - ggplot(data):
+      This method specifies the default data frame to use for the plot, but no aesthetics are defined up front.
+      This is useful when one data frame is used predominantly as layers are added, but the aesthetics may vary
+      from one layer to another.
+
+    - ggplot():
+      This method initializes a skeleton ggplot object which is fleshed out as layers are added. This method is
+      useful when multiple data frames are used to produce different layers, as is often the case in complex
+      graphics.
+
+    ggplot() with no layers defined will produce an error message: "No layers in plot".
 
     Examples
-    ---------
-    >>> import numpy as np
-    >>> import pandas as pd
-    >>> from lets_plot import *
-    >>> x = np.random.uniform(-1, 1, size=100)
-    >>> y = np.random.normal(size=100)
-    >>> dat = pd.DataFrame({'x': x, 'y': 25 * x ** 2 + y})
-    >>> dat['class'] = ['0' if dat['x'][i] < 0 else '1' for i in range(100)]
-    >>> # three ways to invoke ggplot, producing the same output:
-    >>> # (1)
-    >>> ggplot(dat, aes(x='x', y='y')) + layer()
-    >>> # (2)
-    >>> ggplot(dat) + layer()
-    >>> # (3)
-    >>> ggplot() + layer('point', 'identity', dat)
+    --------
+    .. jupyter-execute::
+        :linenos:
+        :emphasize-lines: 12, 14, 16
+
+        import numpy as np
+        import pandas as pd
+        from lets_plot import *
+        LetsPlot.setup_html()
+        np.random.seed(42)
+        n = 100
+        x = np.random.uniform(-1, 1, size=n)
+        y = np.random.normal(size=n)
+        data = pd.DataFrame({'x': x, 'y': 25 * x ** 2 + y})
+        # three ways to invoke ggplot, producing the same output:
+        # (1)
+        ggplot(data, aes(x='x', y='y')) + layer('point')
```

Comment:
Why not just `geom_point()`? Here it is irrelevant, but `geom_point()` looks more familiar.